models/psf.py

```diff
@@ -387,6 +387,13 @@ def get_clip( self, x, y, flux, norm=True, noisy=False, gain=1., rng=None, dtype
         if self.format != 'psfex':
             raise NotImplementedError( "Only know how to get clip for psfex PSF files" )
 
+        if x is None:
+            # x = self.image_shape[1] / 2.
+            x = 0  # I think the x/y are centered on the center of the image
```

Comment:
No... the x and y are the actual image x and y coordinates.  (See docstring on `get_resampled_psf`.) PSFex stores a couple of values `POLZERO1` and `POLZERO2` that are the zero offsets of the polynomial it fits (in FITS coordinates, so 1-offset), and `get_resampled_psf` uses this.

Reply:
So what you are saying is that, for the "representative" PSF of the image, I should default the x,y coordinates to `float( self.header['POLZERO1'] ) - 1` and `POLZERO2`?

Reply:
It's not immediately obvious, but I would probably use the center of the image.  The POLZERO1 nad POLZERO2 aren't necessarily the positions of the "most average" estimate, it's just what it decided to use in its fit.  The commented-out image.shape[n] / 2. are what I'd put in.

Reply:
I assumed we want the center of the image. This is why I've added the size of the image in PSF to begin with. 
But when I set the x,y to the center of the image I got really bad PSFs (sort of donut shaped) and everything failed. 

Now that it is set to zero, I get nice PSFs. I think that those coordinates are not what you think they are.

Reply:
I'm leaving it as-is and making an issue to investigate this.

Reply:
There must be something wrong with the psf solution to your image.

I'm very sure that the coordinates in `get_resampled_psf` are what I think they are.  I've been using versions of that function for a long time.  Look at the code....

But, also, look at `test_psf_rendering` in `tests/models/test_psf.py`.  Uncomment the things that create images (which we should probably replace with an if on whatever conditional variable you've created for interactive tests).  Run the test, and then look first at `/seechange/data/resamp.fits`, which was created with `resamp = psf.get_resampled_psf( 512., 512., dtype=np.float32 )` on a 1024×1024 image.  Next look at `model.fits`, `resid.fits`, and `data.fits`.  data is fake data, with stretched gaussians dropped down. model was built using `PSF.add_psf_to_image`, feeding it the image positions of the psfs that were dropped down, which in turn calls `get_clip` using those image positions.

The problem is somewhere else, not in get_clip.  The best thing to put if x and y are non is the center of the image, not the corner of the image.

What image, and what sequence of functions/commands, did you do that gave you a bad-looking PSF?